make-release.sh

```diff
@@ -184,63 +186,79 @@ replaceFieldSed()
+  git checkout main
+  git pull origin main
+  local HEAD_BRANCH=pr-${MAJOR}.$((MINOR + 1)).x-to-main
+  git checkout -b ${HEAD_BRANCH}
   # Update the version, defined in the antora.yml file, in following keys:
   #    prod-prev-ver-major: "6" [never changes]
   #    prod-ver-major: "7" [never changes]
-  #    prod-prev-ver: "7.24" [always prod-ver - 1]
-  #    prod-ver: "7.25"
+  #    prod-prev-ver: "7.42" [always prod-ver - 1]
+  #    prod-ver: "7.42"
   #    prod-ver-patch: "7.25.2"
   # Major version upgrade is expected to fail.
-  YAMLFILE=antora.yml
+  local YAMLFILE=antora.yml
   # prod-ver should never go down, only up
   versionIsIncremented
   replaceFieldSed "${YAMLFILE}" 'prod-ver-major' "\"${MAJOR}\""
-  replaceFieldSed "${YAMLFILE}" 'prod-ver' "\"${MAJOR}.${MINOR}\""
-  replaceFieldSed "${YAMLFILE}" 'prod-ver-patch' "\"${MAJOR}.${MINOR}.${PATCH}\""
-  replaceFieldSed "${YAMLFILE}" 'prod-prev-ver' "\"${MAJOR}.$((MINOR - 1))\""
+  replaceFieldSed "${YAMLFILE}" 'prod-ver' "\"${MAJOR}.$((MINOR + 1))\""
+  replaceFieldSed "${YAMLFILE}" 'prod-ver-patch' "\"${MAJOR}.$((MINOR + 1)).0\""
+  replaceFieldSed "${YAMLFILE}" 'prod-prev-ver' "\"${MAJOR}.${MINOR}\""
   # Update the version, defined in the antora-playbook-for-publication.yml file, in following keys:
   #    branches: 7.32.x
   #    edit_url: "https://github.com/eclipse/che-docs/edit/7.35.x/{path}"
+  fi  
+  echo "[INFO] Generating single-sourced docs on branch: ${MAIN_BRANCH}."
+  ./tools/checluster_docs_gen.sh
+  ./tools/environment_docs_gen.sh
+  echo "[INFO] Finished handling version update on branch: ${MAIN_BRANCH}."
+  gitPullRequest ${MAIN_BRANCH} ${HEAD_BRANCH}
+}
+
+publicationsBuilderUpdate() {
+  git checkout ${PUBLICATION_BRANCH}
   YAMLFILE=antora-playbook-for-publication.yml
   if [ -f "${YAMLFILE}" ]
   then
-    replaceFieldSed "${YAMLFILE}" 'branches' "\"${MAJOR}.${MINOR}.x\""
-    replaceFieldSed "${YAMLFILE}" 'edit_url' "\"https://github.com/eclipse/che-docs/edit/${MAJOR}.${MINOR}.x/{path}\""
+    sed "/   - branches:/a ${MAJOR}.${MINOR}.x\}" ${YAMLFILE}
   else
-    echo "[WARNING] Cannot find file: ${YAMLFILE} on branch: ${TARGET_BRANCH}. Skipping."
-  fi  
-  echo "[INFO] Generating single-sourced docs on branch: ${TARGET_BRANCH}."
-  ./tools/checluster_docs_gen.sh
-  ./tools/environment_docs_gen.sh
-  echo "[INFO] Finished handling version update on branch: ${TARGET_BRANCH}.."
+    echo "[WARNING] Cannot find file: ${YAMLFILE} on branch: ${MAIN_BRANCH}. Skipping."
+  echo "[INFO] Finished handling version update on branch: ${PUBLICATION_BRANCH}"
+  gitPush ${PUBLICATION_BRANCH}
 }
 
+patchVersionUpdate() {
+  checkoutVersionBranch
+  versionIsIncremented
+  replaceFieldSed "${YAMLFILE}" 'prod-ver-major' "\"${MAJOR}\""
+  replaceFieldSed "${YAMLFILE}" 'prod-ver' "\"${MAJOR}.${MINOR}\""
+  replaceFieldSed "${YAMLFILE}" 'prod-ver-patch' "\"${MAJOR}.${MINOR}.${PATCH}\""
+  replaceFieldSed "${YAMLFILE}" 'prod-prev-ver' "\"${MAJOR}.$((MINOR - 1))\""
+  gitTag
```

Comment:
We should always update the generated docs.

```suggestion
  ./tools/checluster_docs_gen.sh
  ./tools/environment_docs_gen.sh
  gitTag
```